cmd/soroban-rpc/internal/db/event.go

```diff
@@ -97,3 +163,182 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 
 	return nil
 }
+
+type ScanFunction func(
+	event xdr.DiagnosticEvent,
+	cursor Cursor,
+	ledgerCloseTimestamp int64,
+	txHash *xdr.Hash,
+) bool
+
+// trimEvents removes all Events which fall outside the ledger retention window.
+func (eventHandler *eventHandler) trimEvents(latestLedgerSeq uint32, retentionWindow uint32) error {
+	if latestLedgerSeq+1 <= retentionWindow {
+		return nil
+	}
+
+	cutoff := latestLedgerSeq + 1 - retentionWindow
+	_, err := sq.StatementBuilder.
+		RunWith(eventHandler.stmtCache).
+		Delete(eventTableName).
+		Where(sq.Lt{"ledger_sequence": cutoff}).
```

Comment:
there is no `ledger_sequence` column. should this where clause be comparing the cutoff ledger with the id column?

Reply:
oh yes totally forgot about updating this logic 👌